rclpy/rclpy/executors.py

```diff
@@ -384,23 +399,41 @@ def wait_for_ready_callbacks(self, timeout_sec=None, nodes=None):
                         yield handler, srv, node
 
             # Check timeout timer
-            if (timeout_nsec == 0 or
-                    (timeout_timer is not None and timeout_timer.timer_pointer in timers_ready)):
-                break
+            if (
+                timeout_nsec == 0 or
+                (timeout_timer is not None and timeout_timer.timer_pointer in timers_ready)
+            ):
+                raise TimeoutException()
+
+    def wait_for_ready_callbacks(self, *args, **kwargs):
+        """
+        Reuse generator and return callbacks that are ready to be performed.
+
+        See :func:`Executor._wait_for_ready_callbacks` for documentation
+        """
+        if self._cb_iter is None or self._last_args != args or self._last_kwargs != kwargs:
+            # Create a new generator
+            self._last_args = args
+            self._last_kwargs = kwargs
+            self._cb_iter = self._wait_for_ready_callbacks(*args, **kwargs)
+        try:
```

Comment:
Minor nit: I'd put a blank line before this try block, just to visually distinguish it from the if block above.

Reply:
Added in 786b4cf4f9bdc067183eed66f50a0bc1b885b21b